Makefile

```diff
@@ -14,7 +14,55 @@ build:
 	'$(VIRTUALENV_DIR)/bin/pip' install -r ./requirements.txt
 	'$(VIRTUALENV_DIR)/bin/python' setup.py install
 
+# Create a image with docker build & docker run.
+docker-build:
+	# e.g., docker build -t  <image_name> /path/to/Dockerfile/
+	# -t flag - to tag the image with a name so Docker recognizes this name.
+	docker build -t migrate2to3:v1 .
+
+docker-run: docker-build
+	# Runs docker-build part before docker-run part, i.e., docker-run DEPENDS on docker-build.
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 bash
+
+# Create 2 virtualenvs & test if each virtualenv with each python is running correctly.
+docker-py27env:
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 virtualenv py27env -p /usr/bin/python
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 py27env/bin/pip2 install -r requirements.txt
+
+docker-py36env:
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 virtualenv py36env -p /usr/bin/python3
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 py36env/bin/pip3 install -r requirements.txt
+
+test_py2.7: docker-py27env
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py27env/bin/python ./tablesnap --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap -e USER=root migrate2to3:v1 ./py27env/bin/python ./tableslurp --help
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py27env/bin/python ./tablechop --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py27env/bin/python ./tablesnap-sunset --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py27env/bin/python ./tablesnap-sync-manifest --help
+
+test_py3.6: docker-py36env # Note: Expect this to FAIL until Python migration changes done when syntax errors corrected.
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py36env/bin/python ./tablesnap --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap -e USER=root migrate2to3:v1 ./py36env/bin/python ./tableslurp --help
+	docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py36env/bin/python ./tablechop --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py36env/bin/python ./tablesnap-sunset --help
+	# docker run -it --mount type=bind,source=$(PWD),target=/tablesnap migrate2to3:v1 ./py36env/bin/python ./tablesnap-sync-manifest --help
+
+DIRECTORY=/tmp/tablesnap-test
+TIMEOUT_A=5
+TIMEOUT_B=10
+AWS_ACC_KEY_ID := $(shell aws configure get aws_access_key_id)
+AWS_SEC_ACC_KEY := $(shell aws configure get aws_secret_access_key)
+BUCKET=klew-tablesnap-pymigration
```

Comment:
Oh, the default bucket name probably shouldn't have 'klew' in it. That raises a bit of a problem because for each person who runs this script they will need to have their own bucket and the default should reflect that.

I suggest replacing

```
BUCKET=klew-tablesnap-pymigration
```

with

```
BUCKET=$(USER)-tablesnap-tests
```

For you `$(USER)`. will become `klew` on your machine. I'm suggesting `tests` instead of `pymigration` because this VM could be useful for testing lots of different things.

Reply:
Made change.